src/shared/components/Button/Button.style.ts

```diff
@@ -88,3 +90,30 @@ export const StyledText = styled(Text)<TextProps>`
 
   color: inherit;
 `
+
+const BadgeDotStyles = css`
+  &::after {
+    ${square(8)}
+
+    content: '';
+    background-color: ${colors.white};
+    border-radius: 50%;
+    display: block;
+  }
+`
+
+export const Badge = styled.div<{ dot?: boolean }>`
+  ${square(16)}
+
+  font-weight: ${typography.weights.bold};
+  font-size: 0.6rem;
```

Comment:
Let's specify explicit (in px) font size here, since the container size is specified explicitly as well